src/extension.ts

```diff
@@ -68,106 +66,112 @@ function activate(context) {
         indent += ' ';
     }
-    let template = ``;
+    let template: string = ``;
     template += `${indent}/**\n`;
     template += `${indent} * @function `;
 
-    const name = extractFunctionName( signature );
+    const name: string = extractFunctionName( signature );
     template += `${(name === '') ? `{function name}\n` : `${name}\n`}`;
 
-    const parameters = extractParameters( signature );
+    const parameters: string[] = extractParameters( signature );
 
     if ( parameters.length === 0 ) {
         template += `${indent} * @return {type} {description}\n${indent} */\n`;
         return template;
     } else {
-        const prettyParams = prettyParameters( parameters );
-        let parameterString = prettyParams.map( (param) => `${indent} * @param  {type} ${param} {description}\n` );
-        parameterString = parameterString.reduce( ( acc, curr ) => acc.concat( curr ) );
+        const prettyParams: string[] = prettyParameters( parameters );
+        let parameterStrings: string[] = prettyParams.map( (param: string) => `${indent} * @param  {type} ${param} {description}\n` );
+        let parameterString: string = parameterStrings.reduce( ( acc: string, curr: string ) => acc.concat( curr ) ).toString();
         parameterString += `${indent} * @return {type} {description}\n${indent} */\n`;
         return( template.concat( parameterString ) );
     }
-};
+}
 
 /**
  * @function extractParameters
  * @param  {String} signature The function signature
  * @return {List} List of parameters
  */
-const extractParameters = ( signature ) => {
-    const possibleParameters = signature.slice( signature.indexOf( '(' ) + 1, signature.indexOf( ')' ) ).trim();
-    return (possibleParameters === '') ? [] : possibleParameters.split( ',' ).map( (str) => str.trim() );
-};
+function extractParameters( signature: string ): string[] {
+    const possibleParameters: string = signature.slice( signature.indexOf( '(' ) + 1, signature.indexOf( ')' ) ).trim();
+    return (possibleParameters === '') ? [] : possibleParameters.split( ',' ).map( (str: string) => str.trim() );
+}
 
-const addPadding = ( str, max ) => {
+function addPadding( str: string, max: number ): string  {
     if ( str.length < max ) {
-        let i = str.length;
+        let i: number = str.length;
         for ( ; i < max; i++ ) {
             str += ' ';
         }
         return str;
     }
     return str;
-};
+}
 
-const prettyParameters = ( listOfParameters ) => {
-    const lengths = listOfParameters.map( (str) => str.length );
-    const max = Math.max.apply( null, lengths );
+function prettyParameters( listOfParameters: string[] ): string[] {
+    const lengths: number[] = listOfParameters.map( (str: string) => str.length );
+    const max: any = Math.max.apply( null, lengths );
 
-    const mappingFunction = ( x ) => addPadding( x, max );
+    const mappingFunction: (x: string) => string = ( x: string ) => addPadding( x, max );
```

Comment:
This is what I mean.  You have to do `(varname: type) => returnType` for every function declaration similar to this and it can make the line really long.